### PR TITLE
Add confirmation dialog to stop mission button

### LINKED ISSUE #471: Add confirmation popup on the stop button
**Describe the improvement you would like to see**
When clicking the stop button, a confirmation popup should appear like this: 
image

**How will this change existing functionality?**
It will be an extra step to cancel a mission. One can still pause the mission without the popup

**How will this improvement affect the current Threat Model?**
N/A

### PR DESCRIPTION
**Pull request by mrica-equinor**
Closes #471

### CODE REVIEW

File: frontend/src/components/Pages/FrontPage/MissionOverview/StopMissionPopover.tsx
Comment (Eddasol) [inline frontend/src/components/Pages/FrontPage/MissionOverview/StopMissionPopover.tsx:39]:
```suggestion
export const StopMissionDialog = ({ mission }: MissionProps): JSX.Element => {
```

File: frontend/src/components/Pages/FrontPage/MissionOverview/MissionControlButtons.tsx
Comment (Eddasol) [inline frontend/src/components/Pages/FrontPage/MissionOverview/MissionControlButtons.tsx:67]:
```suggestion
                        <StopMissionDialog mission={mission}/>
```

File: frontend/src/components/Pages/FrontPage/MissionOverview/StopMissionDialog.tsx
Comment (Eddasol) [inline frontend/src/components/Pages/FrontPage/MissionOverview/StopMissionDialog.tsx:37]: Duplicated in both StopMissionPopOver.tsx and MissionControlButtons.tsx. Should only be in one of them and imported to the other

File: frontend/src/components/Pages/FrontPage/MissionOverview/StopMissionDialog.tsx
Comment (Eddasol) [inline frontend/src/components/Pages/FrontPage/MissionOverview/StopMissionDialog.tsx:44]:
```suggestion
    const handleDialog = () => {
```

File: frontend/src/components/Pages/FrontPage/MissionOverview/StopMissionDialog.tsx
Comment (Eddasol) [inline frontend/src/components/Pages/FrontPage/MissionOverview/StopMissionDialog.tsx:63]: Why don't we want it to be dismissable?

File: frontend/src/components/Pages/FrontPage/MissionOverview/StopMissionDialog.tsx
Comment (Eddasol) [inline frontend/src/components/Pages/FrontPage/MissionOverview/StopMissionDialog.tsx:50]:
```suggestion
        if (missionId !== mission.id) setIsStopMissionDialogOpen(false)
```

File: frontend/src/components/Pages/FrontPage/MissionOverview/StopMissionDialog.tsx
Comment (mrica-equinor) [reply frontend/src/components/Pages/FrontPage/MissionOverview/StopMissionDialog.tsx:63]: I thought it is would be better to force the user to either cancel or stop, since we have the cancel button which provides the same functionality as dismissible. Do you think we should change it?

Comment (Eddasol) [reply frontend/src/components/Pages/FrontPage/MissionOverview/StopMissionDialog.tsx:63]: We have generally used isDismissable and unless there is a reason to have it as false I would leave it as true for consistency.  If you prefer false that is fine true

Comment (mrica-equinor) [reply frontend/src/components/Pages/FrontPage/MissionOverview/StopMissionDialog.tsx:63]: My reasoning was because in the tablet it can be easy to click somewhere and if you are trying to stop a mission and suddenly click somewhere and have to click the stop button again, that might not be ideal. Depends a bit how crucial are the situations in which the operator is stopping the mission and how urgent. But I'm probably not the best person to answer this question so I'll leave it up to you to decide!

**Review by andchiind — APPROVED** [review]
Seems to work well for me. Once Edda has approved then this can be merged

File: frontend/src/components/Pages/FrontPage/MissionOverview/StopMissionDialog.tsx
Comment (Eddasol) [inline frontend/src/components/Pages/FrontPage/MissionOverview/StopMissionDialog.tsx:55]:
```suggestion
            <Button variant="ghost_icon" onClick={handleDialog}>
```
If the function "handleDialog" doesn't take any arguments this is possible.
Would also consider renaming to "openDialog"

File: frontend/src/components/Pages/FrontPage/MissionOverview/StopMissionDialog.tsx
Comment (mrica-equinor) [reply frontend/src/components/Pages/FrontPage/MissionOverview/StopMissionDialog.tsx:37]: I have changed this and the enum is exported on StopMissiondialog.tsx

File: frontend/src/components/Pages/FrontPage/MissionOverview/StopMissionDialog.tsx
Comment (Eddasol) [inline frontend/src/components/Pages/FrontPage/MissionOverview/StopMissionDialog.tsx:66]:
```suggestion
                        <Typography variant="h5">{TranslateText('Stop mission') + ' ' + mission.name + '?'}</Typography>
```
Translation fail as is now

Comment (Eddasol) [reply frontend/src/components/Pages/FrontPage/MissionOverview/StopMissionDialog.tsx:66]:
```suggestion
                        <Typography variant="h5">{TranslateText('Stop mission')} <strong>'{mission.name}'</strong>? /Typography>
```
Alternatively this

**Pull request merged**
